apis/v1alpha2/tcproute_types.go

```diff
@@ -59,6 +59,18 @@ type TCPRouteStatus struct {
 
 // TCPRouteRule is the configuration for a given rule.
 type TCPRouteRule struct {
+	// Matches add rules for filtering traffic to backends based on addresses
+	// and ports. Any traffic that matches the rule should be routed to the
+	// configured BackendRefs accordingly. Implementations should consider
+	// matches an exclusive configuration: if any traffic does NOT match
+	// according to the given TrafficMatches configuration then the traffic
+	// should not be handled by this route.
+	//
+	// Support: Extended
+	//
+	// +optional
+	Matches *TrafficMatches `json:"matches"`
```

Comment:
One concern I have - what if I want to do this for HTTP/TLS/etc? It seems like today we have HTTPRoute as a superset of TCPRoute; this would be a change from that direction.

In particular, we have a requirement for port matching on HTTP.

Reply:
I did consider this but I was wondering if it might be a bit out of scope considering the conversation thus far in #735, but it seems to me we can just as easily add `TrafficMatches` to `HTTPRouteMatch` as an optional field if we like, no?

Reply:
I think its very reasonable to defer this though

Reply:
:heavy_check_mark: